faker/providers/date_time/__init__.py

```diff
@@ -73,6 +74,21 @@ class ParseError(ValueError):
 
 
 class Provider(BaseProvider):
+    # NOTE: Windows only guarantee second precision, in order to emulate that
+    #       we need to inspect the platform to determine which function is most
+    #       appropriate to generate random seconds with.
+    if platform.system() == "Windows":
+
+        @property
+        def _rand_seconds(self):
+            return self.generator.random.randint
+
+    else:
+
+        @property
+        def _rand_seconds(self):
+            return self.generator.random.uniform
```

Comment:
This could be done with a function that internally uses `platform.system()` to determine which one is the best function, but I do not expect the platform to change at any time when using faker. Making a small effort to avoid checking the same condition over and over seemed worth it - but the formatting turned out a bit wonky somehow.